### PR TITLE
This commit fixes #148: Volume keys don't work on Mac.

### LINKED ISSUE #148: Volume keys don't work on Mac.
The multimedia keys added in #106 don't work on Mac. Apparently you can't just send a volume up event. Here's a possible solution: 

https://github.com/synergy/synergy/issues/3305

### PR DESCRIPTION
**Pull request by gurel**
This commit fixes #148: Volume keys don't work on Mac.
Also added brightness, keyboard illumination, play/pause, next/previous

### CODE REVIEW

**Comment by gurel** [comment]
The failure doesn't seem to be related to the commit. =/

**Comment by octalmage** [comment]
@gurel I restarted the build and it passed! Usually just a connection issue. ANYWAY. 

This looks great! Can't wait to test it.

**Comment by gurel** [comment]
There is also something that can be done, which is to map STOP key to PAUSE since there is no stop key on macs. Although it may give more consistency around multiple operating systems.

File: src/keypress.c [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 #if defined(IS_MACOSX)
 	#include <ApplicationServices/ApplicationServices.h>
+	#import <IOKit/hidsystem/IOHIDLib.h>
+    #import <IOKit/hidsystem/ev_keymap.h>
```
Comment (octalmage) [inline src/keypress.c]: Would you mind fixing this? These should be hard tabs.

File: src/keypress.c [hunk 2]
```diff
@@ -26,6 +28,45 @@
 		 microsleep(DEADBEEF_UNIFORM(0.0, 62.5)))
 #endif
 
+#if defined(IS_MACOSX)
+bool keyCodeRequiresSystemDefinedEvent(MMKeyCode code) {
+    return code == NX_KEYTYPE_SOUND_UP ||
+           code == NX_KEYTYPE_SOUND_DOWN ||
```
Comment (octalmage) [inline src/keypress.c]: This code block too! Sorry to nitpick. :(

**Comment by gurel** [comment]
Hope this update should covers all the hard tab indents. The IDE default of my javascript editor is spaces. Kinda skipped my mind =).

**Comment by octalmage** [comment]
Awesome! I'll get this merged today. Thanks again, this is so great.

**Comment by octalmage** [comment]
Wow this is a ton of fun. How did you come up with this fix? I'm having trouble finding documentation about it. 

Do you know if this will cause any issues on older versions of Mac OS X? Like Yosemite or Mavericks? 

Thanks!

**Comment by octalmage** [comment]
Wow nevermind, IOKit goes back far. I'll get this released!

**Comment by octalmage** [comment]
I found an issue, the keycodes for the multimedia keys overlap with other keys. For example, for the letter "a", keyCodeRequiresSystemDefinedEvent returns 1. I'm going to give them special keycodes so we can convert them.

**Comment by octalmage** [comment]
Here are the keycodes: 

```
NX_KEYTYPE_MUTE: 7
NX_KEYTYPE_PLAY: 16
NX_KEYTYPE_SOUND_UP: 0
NX_KEYTYPE_SOUND_DOWN: 1
NX_KEYTYPE_BRIGHTNESS_UP: 2
NX_KEYTYPE_BRIGHTNESS_DOWN: 3
NX_KEYTYPE_PLAY: 16
NX_KEYTYPE_PREVIOUS: 18
NX_KEYTYPE_NEXT: 17
NX_KEYTYPE_ILLUMINATION_UP: 21
NX_KEYTYPE_ILLUMINATION_DOWN: 22
NX_KEYTYPE_ILLUMINATION_TOGGLE: 23
```

**Comment by octalmage** [comment]
Fixed in #180! Released in RobotJS v0.4.1.

**Comment by gurel** [comment]
Sorry for the late reply, somehow the notification aren't working.

Well it actually wasn't easy to find, but i stumbled to a few solutions:
http://stackoverflow.com/questions/10273159/osx-programmatically-invoke-sound-level-graphic/10520629#10520629
https://github.com/BlueM/cliclick/blob/master/Actions/KeyPressAction.m
https://github.com/SidneyS/specialkey-emulator/blob/master/src/darwin/specialkey-emulator.mm

**Pull request merged**